core/src/wallet/stellar/StellarLikeAccount.cpp

```diff
@@ -0,0 +1,524 @@
+
+                });
+            });
+
+            return eventPublisher->getEventBus();
+        }
+
+        void StellarLikeAccount::startBlockchainObservation() {
+            throw make_exception(api::ErrorCode::IMPLEMENTATION_IS_MISSING, "Not implemented");
+        }
+
+        void StellarLikeAccount::stopBlockchainObservation() {
+            throw make_exception(api::ErrorCode::IMPLEMENTATION_IS_MISSING, "Not implemented");
+        }
+
+        bool StellarLikeAccount::isObservingBlockchain() {
+            throw make_exception(api::ErrorCode::IMPLEMENTATION_IS_MISSING, "Not implemented");
+        }
+
+        std::string StellarLikeAccount::getRestoreKey() {
+            return _params.keychain->getRestoreKey();
+        }
+
+        FuturePtr<ledger::core::Amount> StellarLikeAccount::getBalance() {
+            auto self = getSelf();
+            return async<std::shared_ptr<Amount>>([=] () {
+                const auto& currency = self->getWallet()->getCurrency();
+                soci::session sql(self->_params.database->getPool());
+                stellar::Account account {};
+
+                auto accountId = AccountDatabaseHelper::createAccountUid(self->getWallet()->getWalletUid(), self->_params.index);
+                StellarLikeAccountDatabaseHelper::getAccount(sql, accountId, account);
+                StellarLikeAccountDatabaseHelper::getAccountBalances(sql, accountId, account);
+
+                auto balanceIt = std::find_if(account.balances.begin(), account.balances.end(),
+                        [] (const stellar::Balance& balance) {
+                    return balance.assetType == "native";
+                });
+
+                BigInt amount = (balanceIt != account.balances.end()) ? balanceIt->value : BigInt::ZERO;
+                return std::make_shared<Amount>(self->getWallet()->getCurrency(), 0, amount);
+            });
+        }
+
+        Future<AbstractAccount::AddressList> StellarLikeAccount::getFreshPublicAddresses() {
+            auto self = getSelf();
+            return async<AbstractAccount::AddressList>([=] () {
+                return AbstractAccount::AddressList({self->_params.keychain->getAddress()});
+            });
+        }
+
+        Future<std::vector<std::shared_ptr<api::Amount>>>
+        StellarLikeAccount::getBalanceHistory(const std::string &start, const std::string &end,
+                                              api::TimePeriod precision) {
+            auto self = getSelf();
+            return async<std::vector<std::shared_ptr<api::Amount>>>([=]() -> std::vector<std::shared_ptr<api::Amount>> {
+
+                auto startDate = DateUtils::fromJSON(start);
+                auto endDate = DateUtils::fromJSON(end);
+                if (startDate >= endDate) {
+                    throw make_exception(api::ErrorCode::INVALID_DATE_FORMAT,
+                                         "Start date should be strictly greater than end date");
```

Comment:
"Start date should be strictly **lower** than end date", right ?

Reply:
Or maybe you condition just above is wrong

Reply:
Indeed, I fixed it for all account implementation, they all have this mistake

Reply:
Fixed by fbcda56e6e86857b52b2023d7e3f7d21f4625a02